src/deploy.test.js

```diff
@@ -0,0 +1,102 @@
+const _ = require('lodash')
+const expect = require('chai').expect
+const sinon = require('sinon')
+const deploy = require('./deploy')
+
+const getHarbormaster = (extensions = {}) => _.merge({
+  getEnvironment ({ name }) {
+    return Promise.resolve({
+      name
+    })
+  },
+  postPackage () {
+    return Promise.resolve({
+      id: 'a235ac1b-39cd-460f-ac58-97ce5bc22bdc'
+    })
+  },
+  postRelease () {
+    return Promise.resolve({
+      id: '4d821a35-469b-480f-8f4d-60631d295b09'
+    })
+  }
+}, extensions)
+
+const getContext = (extensions = {}) => _.merge({
+  payload: {
+    comment: {
+      body: '/deploy staging',
+      created_at: '2021-03-11T21:37:02.186Z'
+    },
+    repository: {
+      full_name: 'takescoop/sonar-deploy-action',
+      html_url: 'https://github.com/takescoop/sonar-deploy-action'
+    }
+  }
+}, extensions)
+
+const getConfig = (extensions = {}) => _.merge({
+  defaultEnvironment: 'staging',
+  deploymentManifests: 'test/fixtures/app1.yaml',
+  version: '124cb6d97d86ed4b93eccc95a8ce4ff58a24f843-test',
+  ciUrlPrefix: 'https://app.circleci.com/pipelines/github/',
+  gitRef: '124cb6d97d86ed4b93eccc95a8ce4ff58a24f843',
+  branch: 'feature',
+  trigger: '/deploy',
+  workspace: '.'
+}, extensions)
+
+describe('deploy', () => {
```

Comment:
Also, would be nice to use nock here and generally avoid Sinon where possible

Reply:
My thought is that nock belongs more around testing the harbormaster library itself and in deploy.test.js testing that the client was called correctly. Sinon isn't really doing anything here aside from decorating the test client to assert calls.

Reply:
Yeah on closer inspection, using Sinon to construct your own mock is fine. It's really just the stubbing of shared objects that's bad. Even using stubs to construct the mock methods is fine, as long as the object is a mock you constructed and not part of the source code.